### ISSUE TITLE
Ensure unique form field IDs to fix label focus, JS behavior, a11y, and spec compliance in new corpus search component

### ISSUE THREAD

**Issue by elisa-a-v** [freelawproject/courtlistener#6198]
### Describe the issue.

Our current implementation of the corpus search in new templates is producing duplicate id attributes in the DOM, which is a clear spec violation.

### Current implementation

- We render the entire form more than once on the page, so every single field is present at least twice in the document.
- Within each form, we have multiple fieldsets (for different search scopes), and some fields are present in more than one scope (for example `case_name` and `docket_number` are present in all scopes but judges).
- Within a form, only one fieldset is active at a time; inactive ones are disabled and hidden, but the elements are still present in the DOM.

### Requirement

We need a way to ensure each input in the document has a unique ID, and that every label is correctly associated with its input.

### Why this matters

Markup is incorrect:
- Duplicate IDs are explicitly invalid HTML.

It hurts UX:
- Labels with `for` attributes don’t consistently focus their corresponding input.
- Screen readers cannot reliably map labels to inputs. This impacts a11y in searches across **all** pages, so one of CourtListener's core features could be rendered virtually unusable in some situations for a lot of people.

Maintainability is also hurt:
- `document.getElementById()` returns only the first match, breaking JS that expects a unique element.
- CSS selectors like `#field_id` apply unpredictably across duplicates. With Tailwind this is probably not too bad though.
- Tests may target the wrong element.

### Possible solutions

- Django's Forms have an `auto_id` but I think we can only tweak the id by having multiple form instances. We currently use the same one everywhere, but we could introduce per-scope forms and setting unique ids that way.
- `widget_tweaks` provides a way to override attributes in the template. This sounds like it could be the simplest? But it looks like we don't actually have it installed even though it's being loaded in a couple of templates.
- Using some sort of scope namespacing. Maybe updating the IDs whenever the scope changes, so only the inputs in the active fieldset have `id` and `for` set. We're already using JS to handle the fieldsets on scope change.
- Using the very same DOM nodes everywhere and just moving them around. Semantically this makes sense to me, but sounds cumbersome to implement.
- Something else?

### What type of issue is this?

To Do

**Comment by mlissner** [comment]
I'm not super attuned to this problem, but one question I have is whether unique IDs will affect the GET parameters that the form generates. I assume it will, right? If that's the case, that's going to be a big issue since we use the same parameters in the API.

> Using the very same DOM nodes everywhere and just moving them around. Semantically this makes sense to me, but sounds cumbersome to implement.

I don't understand this one? Maybe some notional HTML would help me?

**Comment by mlissner** [comment]
For Metallica, just research.
